docs/reference/stdlib.md

```diff
@@ -0,0 +1,660 @@
+
+`countFastq()`
+: Counts the number of records in a [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) file. See the {ref}`operator-splitfastq` operator for available options.
+
+`countJson()`
+: Counts the number of records in a JSON file. See the {ref}`operator-splitjson` operator for available options.
+
+`countLines()`
+: Counts the number of lines in a text file. See the {ref}`operator-splittext` operator for available options.
+
+`error( message = null )`
+: Throw a script runtime error with an optional error message.
+
+`exit( exitCode = 0, message = null )`
+: :::{deprecated} 22.06.0-edge
+  Use `error()` instead
+  :::
+: Stop the pipeline execution and return an exit code and optional error message.
+
+`file( filePattern, options = [:] )`
+: Get one or more files from a path or glob pattern. Returns a [Path](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html) or list of Paths if there are multiple files.
+
+: The following options are available:
+
+  `checkIfExists`
+  : When `true`, throws an exception if the specified path does not exist in the file system (default: `false`)
+
+  `followLinks`
+  : When `true`, follows symbolic links when traversing a directory tree, otherwise treats them as files (default: `true`)
+
+  `glob`
+  : When `true`, interprets characters `*`, `?`, `[]` and `{}` as glob wildcards, otherwise handles them as normal characters (default: `true`)
+
+  `hidden`
+  : When `true`, includes hidden files in the resulting paths (default: `false`)
+
+  `maxDepth`
+  : Maximum number of directory levels to visit (default: *no limit*)
+
+  `type`
+  : Type of paths returned, can be `'file'`, `'dir'` or `'any'` (default: `'file'`)
+
+: See also: {ref}`Channel.fromPath <channel-path>`.
+
+`files( filePattern, options = [:] )`
+: Convenience method for `file()` that always returns a list.
+
+`groupKey( key, size )`
+: Create a grouping key to use with the {ref}`operator-grouptuple` operator.
+
+`multiMapCriteria( closure )`
+: Create a multi-map criteria to use with the {ref}`operator-multiMap` operator.
+
+`sendMail( params )`
+: Send an email. See {ref}`mail-page`.
+
+`splitCsv()`
+: Splits a CSV file into a list of records. See the {ref}`operator-splitcsv` operator for available options.
+
+`splitFasta()`
+: Splits a [FASTA](https://en.wikipedia.org/wiki/FASTA_format) file into a list of records. See the {ref}`operator-splitfasta` operator for available options.
+
+`splitFastq()`
+: Splits a [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) file into a list of records. See the {ref}`operator-splitfastq` operator for available options.
+
+`splitJson()`
+: Splits a JSON file into a list of records. See the {ref}`operator-splitjson` operator for available options.
+
+`splitText()`
+: Splits a text file into a list of lines. See the {ref}`operator-splittext` operator for available options.
```

Comment:
Note to self: these are `Path` extension methods, should be listed in that section. Same for `count*` methods